### PR TITLE
[GH-334] Refactor every game update (tick) action 

### PR DESCRIPTION
**Pull request by Nico-Sanchez**
1st iteration of #334 

Game update handler was messy. This PR wants to expose each action to see refactor oportunities with more clarity.

- Moves out every tick action to a separate function.
- Each of the mentioned functions have to receive and return the game state map.

### CODE REVIEW

File: apps/arena/lib/arena/game_updater.ex [hunk 1]
```diff
@@ -557,12 +505,208 @@ defmodule Arena.GameUpdater do
...
+
+  # Remove exploded projectiles
+  # Update expired to explode status
+  defp update_projectiles_status(%{projectiles: projectiles} = game_state) do
+    updated_projectiles =
+      projectiles
+      |> Map.reject(fn {_projectile_id, projectile} ->
+        projectile.aditional_info.status == :EXPLODED
+      end)
+      |> Enum.reduce(%{}, fn {projectile_id, projectile}, acc ->
```
Comment (AminArria) [inline apps/arena/lib/arena/game_updater.ex:545]: You are iterating twice over `projectiles`, should we do this both things in `Enum.reduce/3`?

Sort of like
```elixir
Enum.reduce(projectiles, projectiles, fn {projectile_id, projectile}, acc ->
  case projectile.aditional_info.status do
    :EXPLODED -> Map.delete(acc, projectile_id)
    :EXPIRED -> put_in(acc, [projectile_id, :aditional_info, :status], :EXPLODED))
    _ -> acc
  end
end)

File: apps/arena/lib/arena/game_updater.ex [hunk 2]
```diff
@@ -557,12 +505,208 @@ defmodule Arena.GameUpdater do
...
+           players: players,
+           obstacles: obstacles,
+           external_wall: external_wall,
+           ticks_to_move: ticks_to_move
+         } = game_state
+       ) do
+    # We don't want to move recently exploded projectiles
+    {recently_exploded_projectiles, alive_projectiles} =
+      projectiles
+      |> Enum.split_with(fn {_projectile_id, projectile} ->
```
Comment (AminArria) [inline apps/arena/lib/arena/game_updater.ex:576]: You can do `Map.split_with/2` with almost the same result, except the tuple will contain maps so you don't need the `Map.new/1` later on

**Review by AminArria — CHANGES_REQUESTED** [review]

File: apps/arena/lib/arena/game_updater.ex (hunk 2 above)
Comment (Nico-Sanchez) [reply apps/arena/lib/arena/game_updater.ex:576]: Nice catch! I always forget that Map has some enum functions 😅

File: apps/arena/lib/arena/game_updater.ex (hunk 1 above)
Comment (Nico-Sanchez) [reply apps/arena/lib/arena/game_updater.ex:545]: Yeah, looks much better. Done!

**Review by agustinesco — DISMISSED** [review]
LGTM! i tested this pr with multiple players, projectiles and mechanics and everythoing worked fine! maybe we could deploy this and test it with multiple players online!

**Review by manucamejo — APPROVED** [review]

**Comment by manucamejo** [comment]
@Nico-Sanchez I think that you can take outside the assigns to the new game from lines `500`, `501` and `502` in `game_updater.ex`

**Comment by Nico-Sanchez** [comment]
> @Nico-Sanchez I think that you can take outside the assigns to the new game from lines `500`, `501` and `502` in `game_updater.ex`

Nice catch! Done!

**Review by manucamejo — APPROVED** [review]

**Review by AminArria — APPROVED** [review]

**Pull request merged**